homeassistant/config_entries.py

```diff
@@ -817,13 +817,25 @@ def async_get_options_flow(config_entry: ConfigEntry) -> "OptionsFlow":
         raise data_entry_flow.UnknownHandler
 
     @callback
-    def _abort_if_unique_id_configured(self) -> None:
+    def _abort_if_unique_id_configured(
+        self, update_entry: Dict[Any, Any] = None
+    ) -> None:
         """Abort if the unique ID is already configured."""
         if self.unique_id is None:
             return
 
-        if self.unique_id in self._async_current_ids():
-            raise data_entry_flow.AbortFlow("already_configured")
+        for entry in self._async_current_entries():
+            if entry.unique_id == self.unique_id:
+                if (
+                    self.hass is not None
+                    and update_entry is not None
+                    and not update_entry.items() <= entry.data.items()
```

Comment:
Let's skip this check, it's up to the caller to verify that it's different, else shouldn't have passed.

Reply:
I really do think we should not do that, and allow for providing partial data that gets merged. The main usage for this functionality would probably be updating existing entries based on discovery data, which most likely is incomplete (and normally added by doing the full flow).

As an example (which caused this PR in the first place), Hue has the linking part in its flow, so the username is missing at the discovery point.

As a workaround, the integration would be fetching the current options, copy it, and put it in again. I'm afraid that would be recurring logic in config flows supporting discovery.

Reply:
You're right. I'm wrong. Let's keep it.